src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -136,14 +136,30 @@ export function start(manifest) {
 
 The following methods are provided:
 
-##### `app.render(request)`
+##### `app.render(request, routeData, locals)`
 
-This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes, that do not render pages.
+This method calls the Astro page that matches the request, renders it, and returns a Promise to a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object. This also works for API routes that do not render pages.
 
 ```js
 const response = await app.render(request);
 ```
 
+The method accepts a mandatory `request` argument, and two other optionals arguments: [`routeData`](en/reference/integrations-reference/#routedata-type-reference) and [`locals`](/en/guides/middleware/#locals).
+
+The example below reads a header named `x-private-header`, which attempts to parse it as an object and pass it to `locals`, which can then be passed to any [middleware function](/en/guides/middleware/).
```

Comment:
```suggestion
The example below reads a header named `x-private-header`, attempts to parse it as an object and pass it to `locals`, which can then be passed to any [middleware function](/en/guides/middleware/).
```
I think the which is out of place, but otherwise, I think this works! 

The only other thing we could do is explain what's going on with `routeData`, and why it's `undefined` in the arguments in the example? (e.g. is it a placeholder because `locals` has to come third? Is it actually important that it have the value undefined in this example?)

Reply:
@ematipico Did you have a chance to see this comment?